src/gluonts/dataset/split.py

```diff
@@ -322,11 +334,12 @@ def training_entry(self, entry: DataEntry) -> DataEntry:
     def test_pair(
         self, entry: DataEntry, prediction_length: int, offset: int = 0
     ) -> Tuple[DataEntry, DataEntry]:
-        date = self.date.asfreq(entry[FieldName.START].freq)
         return (
-            TimeSeriesSlice(entry, prediction_length)[: date + offset],
+            TimeSeriesSlice(entry, prediction_length)[: self.date + offset],
             TimeSeriesSlice(entry, prediction_length)[
```

Comment:
I would actually get rid of the `TimeSeriesSlice` class, but maybe in a separate PR (so this one is not breaking)

Reply:
+1